Update population_dynamics.rst (#658)
lumicks/pylake: docs/tutorial/population_dynamics.rst

This description is the most basic implementation of an HMM. For detailed theoretical and practical
explanations on the use and training of HMMs see the seminal work by Rabiner :cite:`Rabiner1989HMM`.

Fit an HMM using the Baum-Welch algorithm::

    hmm = lk.HiddenMarkovModel(force["0s":"2s"], n_states=2)

This algorithm estimates the model parameters based on the provided data.
One of these estimated parameters is the transition matrix which can be extracted as::

>     >>> hmm.transition_matrix
>     array([[0.96339315, 0.03660685],
>        [0.08215035, 0.91784965]])

Reviewer: This is very minor, but I think typically, there are a few more spaces in the repl output which help align the matrix visually.

```suggestion
    >>> hmm.transition_matrix
    array([[0.96339315, 0.03660685],
           [0.08215035, 0.91784965]])
```

This description is the most basic implementation of an HMM. For detailed theoretical and practical
explanations on the use and training of HMMs see the seminal work by Rabiner :cite:`Rabiner1989HMM`.

> Run the Baum-Welch algorithm to obtain the parameters of the Hidden Markov Model, such as determine the transition matrix:

Reviewer: There seems to be a typo here? Also, I wonder whether it can be a little confusing, since the line of code presented here doesn't actually return the parameters itself, but a model.

I do agree that it's nice to mention the algorithm, in case people are interested. Maybe something like?

```suggestion
Fit an HMM using the Baum-Welch algorithm::
```

And then mention later that the estimated parameters can be extracted (for instance, the transition matrix is available through `hmm.transition_matrix`).

```rst
This algorithm estimates the model parameters based on the provided data.
One of these estimated parameters is the transition matrix which can be extracted as::

    >>> hmm.transition_matrix
```

Reviewer: Not sure if this was intentional, but did you decide that the second part (mentioning the transition matrix being estimated and how to obtain it) is not so useful?

Author: I actually think mentioning the transition matrix is very useful. I thought I just committed your suggestions, but may have missed this one. I did not have time yet to carefully look at the PR and finalize it.

Reviewer: Ahh okay, I wasn't sure whether it was ready for rereview yet. Since the comments were resolved I thought you might have forgotten to rerequest review. But yeah, take your time 👍

    >>> print(hmm.fit_info)
    HmmFitInfo(converged=True, n_iter=5, bic=1589.2881620885353, aic=1550.067858853102, log_likelihood=-768.033929426551)

The means for each of the states can be accessed as::

    >>> hmm.means

>     array([ 8.79212579, 10.08962247])

Reviewer: 👍

.. image:: figures/population_dynamics/hmm_hist.png

> Apply the Viterbi algorithm to assign a state to each data point and obtain the temporal evolution of the fit::

Reviewer: ```suggestion
Apply the Viterbi algorithm to assign the most probable state to each data point and obtain the temporal evolution of the fit::
```
